File: Application/Services/UserService.cs
using Core.Entities;
using Core.Interfaces;

namespace Application.Services
{
    public class UserService
    {
        private readonly IUserRepository _userRepository;

        public UserService(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public async Task<User> GetUserById(string id)
        {
            return await _userRepository.GetUserById(id);
        }

        public async Task<IEnumerable<User>> GetAllUser()
        {
            return await _userRepository.GetAllUser();
        }

        public async Task CreateUser(User user)
        {
            await _userRepository.CreateUser(user);
        }

        public async Task UpdateUser(User user)
        {
            await _userRepository.UpdateUser(user);
        }

        public async Task DeleteUser(string id)
        {
            await _userRepository.DeleteUser(id);
        }
    }
}

File: ui/src/services/userService.ts
// src/services/userService.ts
import axios from "axios";
import { User } from "@/types"; // Asegúrate de tener un tipo User definido en src/types

const API_URL = "http://localhost:5000/api/users"; // Reemplaza con la URL de tu API

export const getAllUsers = async (): Promise<User[]> => {
  const response = await axios.get(API_URL);
  return response.data;
};

export const getUserById = async (id: string): Promise<User> => {
  const response = await axios.get(`${API_URL}/${id}`);
  return response.data;
};

export const createUser = async (user: User): Promise<void> => {
  await axios.post(API_URL, user);
};

export const updateUser = async (id: string, user: User): Promise<void> => {
  await axios.put(`${API_URL}/${id}`, user);
};

export const deleteUser = async (id: string): Promise<void> => {
  await axios.delete(`${API_URL}/${id}`);
};
